olp-cpp-sdk-core/tests/thread/TaskContinuationTest.cpp

```diff
@@ -86,44 +82,28 @@ TEST_F(TaskContinuationFixture, FinallyNotSet) {
 }
 
 TEST_F(TaskContinuationFixture, CancelBeforeRun) {
-  std::promise<void> promise;
+  std::promise<ResponseType<olp::client::HttpResponse>> promise;
   auto future = promise.get_future();
 
-  olp::client::ApiResponse<olp::client::HttpResponse, olp::client::ApiError>
-      result;
-
-  auto cont =
-      olp::thread::TaskContinuation(std::move(scheduler))
-          .Then([](olp::thread::ExecutionContext,
-                   std::function<void(int)> next) { next(1); })
-          .Finally([&promise, &result](
-                       olp::client::ApiResponse<int, olp::client::ApiError>
-                           response) {
-            result = response.GetError();
-            promise.set_value();
-          });
+  auto cont = olp::thread::TaskContinuation(std::move(scheduler))
+                  .Then([](olp::thread::ExecutionContext,
+                           std::function<void(int)> next) { next(1); })
+                  .Finally([&promise](ResponseType<int> response) {
+                    promise.set_value(response.GetError());
+                  });
 
   cont.CancelToken().Cancel();
 
   cont.Run();
 
-  EXPECT_EQ(future.wait_for(std::chrono::milliseconds(1000)),
-            std::future_status::ready);
-
-  future.get();
-
-  EXPECT_FALSE(result.IsSuccessful());
-  EXPECT_EQ(result.GetError().GetHttpStatusCode(),
-            static_cast<int>(olp::http::ErrorCode::CANCELLED_ERROR));
+  EXPECT_EQ(future.get().GetError().GetErrorCode(),
```

Comment:
Please use the approach described above.

Please also check if the response is not successful. And also do the same in all the similar cases

Reply:
done